File: src/python/tutorias/domain/reserva.py
from datetime import date

from .estado_reserva import EstadoReserva


class Reserva:
    def __init__(self, id_reserva: int, fecha_reserva: date):
        self._id = id_reserva
        self._estado = EstadoReserva.PENDIENTE
        self._fecha_reserva = fecha_reserva

    def confirmar(self) -> None:
        self._estado = EstadoReserva.CONFIRMADA

    def cancelar(self) -> None:
        self._estado = EstadoReserva.CANCELADA

    def reprogramar(self) -> None:
        self._estado = EstadoReserva.PENDIENTE

    def marcar_realizada(self) -> None:
        self._estado = EstadoReserva.REALIZADA

File: src/java/edu/uees/tutorias/domain/Reserva.java
package edu.uees.tutorias.domain;

import java.time.LocalDate;

public class Reserva {

    private int id;
    private EstadoReserva estado;
    private LocalDate fechaReserva;

    public Reserva(int id, LocalDate fechaReserva) {
        this.id = id;
        this.fechaReserva = fechaReserva;
        this.estado = EstadoReserva.PENDIENTE;
    }

    public void confirmar() {
        estado = EstadoReserva.CONFIRMADA;
    }

    public void cancelar() {
        estado = EstadoReserva.CANCELADA;
    }

    public void reprogramar() {
        estado = EstadoReserva.PENDIENTE;
    }

    public void marcarRealizada() {
        estado = EstadoReserva.REALIZADA;
    }

    public int getId() {
        return id;
    }

    public EstadoReserva getEstado() {
        return estado;
    }
}